index.js

```diff
@@ -134,21 +134,55 @@ exports.createServer = function (opts) {
       ogr.timeout(opts.timeout)
     }
 
-    let format = req.body.format || 'shp'
+    let format = req.body.format.toLowerCase() || 'shp'
+
+    ogr.format(format)
+
+    const sendResponse = (buf) => {
 
-    ogr.format(format).exec(function (er, buf) {
-      if (isOgreFailureError(er))
-        return res
-          .status(400)
-          .json({errors: er.message.replace('\n\n', '').split('\n')})
-      if (er) return next(er)
       res.header('Content-Type', 'application/zip')
       res.header(
         'Content-Disposition',
         'filename=' + (req.body.outputName || 'ogre.zip')
       )
       res.end(buf)
-    })
+
+    }
+
+    try {
+
+      switch (format) {
+        // These formats must use .destination
+        case 'dxf':
+        case 'dgn':
+        case 'txt':
+        case 'gxt':
+        case 'gmt':
+          
+          // Random string to prevent multiple request being overwritten
+          let randomId = Math.random().toString(36).substring(7);
+
+          let tmpDestination = join(__dirname, `/tmp-${randomId}.${format}`)
+          await ogr.destination(tmpDestination).promise()
+
+          let bufD = await fs.promises.readFile(tmpDestination, 'utf8')
+          fs.unlink(tmpDestination, noop)
```

Comment:
Can this error be handled rather than `noop`ed away?  I think it could be promisified and awaited.

Reply:
What do you think would be the appropriate action if there is an error deleting the temporary file? Abort the entire operation by throwing an error? Even if the conversion process was successful?

Reply:
I would abort the operation.  If the file has some strange issues where it cannot be deleted, how are we sure the operation was really successful?